packages/lesswrong/components/posts/UserSuggestNominations.tsx

```diff
@@ -1,33 +1,131 @@
+    ErrorBoundary,
+    PostsByVoteWrapper,
+    ReadHistoryTab,
+    PostsListUserCommentedOn,
+    Typography,
+    LWTooltip,
+    AllPostsPage
+  } = Components
+
+  // Handle url-encoded special case, otherwise parseInt year
+  const before2020 = '≤2020'
+  const year = [before2020, '%e2%89%a42020'].includes(params?.year) ?
+    before2020 :
+    parseInt(params?.year)
+  if (!currentUser) return null
+
+  const startDate = year === before2020 ? undefined : new Date(year, 0, 1)
+  const endDate = year === before2020 ? new Date(2020, 0, 1) : new Date(year + 1, 0, 1)
+
+  const handleChangeTab = (e: React.ChangeEvent, value: string) => {
+    navigate({
+      ...location,
+      search: `?tab=${value}` + 
+        (value === 'all' ? `&after=${(dateStr(startDate))}&before=${dateStr(endDate)}&timeframe=yearly` : ''),
+    })
+  }
+
+  const activeTab = query.tab || 'votes'
+  return <AnalyticsContext pageContext="nominationPage">
+    <SingleColumnSection>
+      <Typography variant="display2" className={classes.headline}>
+        {preferredHeadingCase(`Nominate Posts for ${year} review`)}
+      </Typography>
+
+      <Tabs
+        value={activeTab}
+        onChange={handleChangeTab}
+      >
+        <Tab 
+          className={classes.tab} 
+          value="votes" 
+          label={
+          <LWTooltip title={`Posts from ${year} you've upvoted`}>Voted On</LWTooltip>}
+        />
+        <Tab
+          className={classes.tab}
+          value="comments"
+          label={<LWTooltip title={`Posts from ${year} you've commented on`}>Commented On</LWTooltip>}
+        />
+        <Tab
+          className={classes.tab}
+          value="read"
+          label={<LWTooltip title={`Posts from ${year} you've read`}>Read</LWTooltip>}
+        />
+        <Tab
+          className={classes.tab}
+          value="all"
+          label={<LWTooltip title={`All posts from ${year}`}>All Posts</LWTooltip>}
+        />
+      </Tabs>
+
+      {(activeTab === 'votes') && <>
+        <SectionTitle title={`Your Strong Upvotes for posts from ${year}`}/>
         <PostsByVoteWrapper voteType="bigUpvote" year={year}/>
-      </SingleColumnSection>
-      <SingleColumnSection>
-        <SectionTitle title={`Your Upvotes from ${year}`}/>
+        <SectionTitle title={`Your Upvotes for posts from ${year}`}/>
         <PostsByVoteWrapper voteType="smallUpvote" year={year}/>
-      </SingleColumnSection>
-    </ErrorBoundary>
+      </>}
+
+      {activeTab === 'comments' && <>
+        <SectionTitle title={`Posts from ${year} you've commented on`}/>
+        <PostsListUserCommentedOn filter={{startDate, endDate, showEvents: false}} sort={{karma: true}}/>
+      </>}
+
+      {activeTab === 'read' && <>
+        <SectionTitle title={`Posts from ${year} you've read`}/>
+        <ReadHistoryTab groupByDate={false} filter={{startDate, endDate, showEvents: false}} sort={{karma: true}}/>
+      </>}
+
+      {activeTab === 'all' && <AllPostsPage/>}
```

Comment:
Does this... work?

Reply:
<img width="973" alt="image" src="https://github.com/user-attachments/assets/09ea87d4-86f8-4719-93ba-e671aacf2b2c">
🙃